services/rsources/stats_collector.go

```diff
@@ -39,6 +39,14 @@ type StatsCollector interface {
 	// so that all necessary indices can be created, since a JobStatus
 	// doesn't carry all necessary job metadata such as jobRunId, taskRunId, etc.
 	JobStatusesUpdated(jobStatuses []*jobsdb.JobStatusT)
+
+	// SkipFailedRecords skips the collection of failed records
+	//
+	// Any further calls to `JobsStoredWithErrors`, `BeginProcessing` and `JobStatusesUpdated`
+	// will not collect failed records
+	//
+	// Make sure required failed records are collected before calling this method
+	SkipFailedRecords()
```

Comment:
This looks more like a producer Option rather than an interface method. What are we trying to achieve? Can we find a solution that doesn't involve a behaviour-changing method (method that affects the behaviour of other methods)?

Reply:
fair enough

But the interface already has something like that in `BeginProcessing`.

Reply:
`BeginProcessing` is for marking the start of collection, since you need to capture job metadata before you can process job statuses. On the contrary, mutating the behaviour of the collector at an arbitrary stage can lead to undesired behaviour and/or bugs to get introduced.

Decoupling collection failed records from collection of other stats could be the way to go if we don't want to have stats collector aware of the special drain error code `410`.